extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -277,6 +278,12 @@ func (a *genericActuator) deployMachineDeployments(ctx context.Context, cluster
 			},
 		}
 
+		// Find the workerpool related to current machine deployment.
+		workerPool := workerPoolForMachineDeployment(worker, deployment)
+		if workerPool == nil {
+			a.logger.Error(fmt.Errorf("Error finding worker pool for machine deployment"), "Worker pool settings will not be applied on related machine deployment.", "worker", fmt.Sprintf("%s/%s", worker.Namespace, worker.Name))
```

Comment:
I guess its the "fault" of the  extension provider if machine deployments are generated that do not include the worker pool name, but I have troubles with just logging and continuing here. 
Can we have at least an event on the worker resource here?
Possibly it makes even sense to fail the reconciliation as otherwise the user thinks that all settings have been successfully applied - but in fact they are not.

Reply:
Yes, agreed. It makes to fail the reconciliation, to avoid confusion. I'll make that change.

Reply:
This code-path is removed based on this comment: https://github.com/gardener/gardener/pull/2563#discussion_r457007752 .